effectful/handlers/llm/template.py

```diff
@@ -4,9 +4,25 @@
 from collections import ChainMap
 from collections.abc import Callable, Mapping, MutableMapping
 from dataclasses import dataclass
-from typing import Any
+from typing import Annotated, Any
 
-from effectful.ops.types import INSTANCE_OP_PREFIX, NotHandled, Operation
+from effectful.ops.types import INSTANCE_OP_PREFIX, Annotation, NotHandled, Operation
+
+
+class _IsRecursiveAnnotation(Annotation):
+    @classmethod
+    def infer_annotations(cls, sig: inspect.Signature) -> inspect.Signature:
+        return sig
```

Comment:
Can you add an assertion in here that `IsRecursive` does not appear on any of the parameters in the signature? It should only be used on the return type.